handlers/discount-api/src/discountEndpoint.ts

```
@@ -104,10 +104,10 @@ const getDiscountPreview = async (
 		discount.productRatePlanId,
 	);
 
-	if (!previewResponse.success || previewResponse.invoiceItems.length != 2) {
+	if (!previewResponse.success || previewResponse.invoiceItems.length < 2) {
```

supporter plus has 2 charges (base plus contribution)
I wasn't sure how valuable this check was/what it's protecting us from? but I kept it in a loose form.  Should I keep it in the limit 2->3 or is minimum 2 ok?  Or remove completely?

---

funny thing is it only supported digi sub before which has one charge.
So the 2 referred to the single digi sub charge plus the new discount charge.

Now it can be 3 as we have S+ with two charges plus one new discount charge.

Maybe we could look at the subscription and count the charges, and then check for n+1 items in the preview, but I'm not 100% sure what that check was intended to assert.  Perhaps it's just checking that the new charge is being planned to trigger on the right date, otherwise it wouldn't appear in the invoice?

@rupertbates any thoughts?

---

I think it's just following the principle of 'set out your expectations and if the situation you're in doesn't match them fail at that point rather than trying to fudge it'. Seems like a reasonable thing to do to me but if it becomes overly complicated when you add S+ into the mix I wouldn't be too upset to lose it.

---

I see, so fail fast and loudly principle rather than leaving zuora in a mess.  I will think about whether to count the charges (on non removed rate plans) later.